### ISSUE TITLE
KeepAlive Option for SFTP Connection

### ISSUE THREAD

**Issue by akhon** [rclone/rclone#4353]
#### What is your current rclone version (output from `rclone version`)?
1.52.1

#### What problem are you are trying to solve?
Get KeepAlive option available in rclone: http://man.openbsd.org/ssh_config#ServerAliveInterval
#### How do you think rclone should be changed to solve that?
new few options: interval and number (?)

**Comment by ncw** [comment]
I think this would be a reasonably simple patch.

It would start a go routine off a timer, then every interval it would
- get the number of conneections in the pool
- for that many connections
    - call getSftpConnection
    - call putSftpConnection with an error `errors.New("connection keepalive failed")`
- possibly after the run if there are no connections left in the pool add one?

The rest of the logic, testing the connections etc is there already.

Would you like to have a go at this?

**Comment by akhon** [comment]
@ncw yeah. sounds good. let me know when we can test this, so we can verify that keep-alive works as expected, we have sandbox setup pre-configured here!

**Comment by akhon** [comment]
How is it going?

**Comment by akhon** [comment]
friendly ping!

**Comment by ncw** [comment]
I haven't had time to work on this, alas. Is it something you'd like to do @akhon ? There is a description of what needs doing above: https://github.com/rclone/rclone/issues/4353#issuecomment-644381669

**Comment by TaleOfThor** [comment]
How is it going?

**Comment by ncw** [comment]
I think we are looking for someone to implement this. Interested @TaleOfThor ?
